### PR TITLE
fix: bifrost xcm types 

### PR DESCRIPTION
**Pull request by Asmadek**
* get types for xcm versioned from call arguments

### CODE REVIEW

**Comment by Asmadek** [comment]
Task linked: [CU-8693fvkg5 Add new XCM type - StagingXcmVersionedMultiAsset](https://app.clickup.com/t/8693fvkg5)

File: src/renderer/shared/lib/utils/substrate.ts [hunk 1]
```diff
@@ -3,8 +3,10 @@ import { BaseTxInfo, getRegistry, GetRegistryOpts, OptionsWithMeta, TypeRegistry
 import { isHex, hexToU8a, bnMin, BN_TWO, BN } from '@polkadot/util';
 import { blake2AsHex } from '@polkadot/util-crypto';
 
-import { Address, CallData, CallHash } from '@shared/core';
+import { Address, CallData, CallHash, XcmPallets } from '@shared/core';
+// TODO: Incorrect import
```
Comment (tuul-wq) [inline src/renderer/shared/lib/utils/substrate.ts:7]: All of these constants are used only in `shared/utils/substrate` we can safely move to `shared/utils/constants`
```javascript
export const THRESHOLD = BN_THOUSAND.div(BN_TWO);
export const DEFAULT_TIME = new BN(6_000);
export const ONE_DAY = new BN(24 * 60 * 60 * 1000);
```

File: src/renderer/shared/lib/utils/substrate.ts [hunk 2]
```diff
@@ -128,3 +130,35 @@ export const getTypeVersions = (api: ApiPromise, typeName: string): string[] =>
...
+  api: ApiPromise,
+  transferType: XcmTransferType,
+): { pallet: XcmPallets; call: string } => {
+  if (transferType === 'xtokens') {
+    return { pallet: 'xTokens', call: 'transferMultiasset' };
+  }
+
+  const pallet = api.tx.xcmPallet ? 'xcmPallet' : 'polkadotXcm';
+
+  if (transferType === 'xcmpallet') {
```
Comment (tuul-wq) [inline src/renderer/shared/lib/utils/substrate.ts:154]: `const enum XcmTransferType` would be a nice enhancement, what do you think?
To remove string conditions

**Review by tuul-wq — COMMENTED** [review]
Few comments

**Review by tuul-wq — APPROVED** [review]

**Review by stepanLav — APPROVED** [review]

**Pull request merged**